RadixWallet/Core/SharedModels/Assets/OnLedgerEntity.swift

```diff
@@ -336,13 +336,13 @@ extension OnLedgerEntity {
 
 		public let resourceAddress: ResourceAddress
 		public let atLedgerState: AtLedgerState
-		public let amount: RETDecimal
+		public var amount: ResourceAmount
```

Comment:
The main piece of the implementation, the ResourceAmount which contains FiatWorth and nominal amount is kept in OnLedgerEntity entities such that is very easy to control the tokenPrice, currency or visibility update from one place - AccountPortfolio.
I don't like this 100%, but alternative would have been to create a mirroring DTOs with additional FiatWorth field.